src/astro/databases/base.py

```diff
@@ -461,10 +464,32 @@ def load_file_to_table_natively(
         source_file: File,
         target_table: Table,
         if_exists: LoadExistStrategy = "replace",
+        native_support_kwargs: Optional[Dict] = None,
         **kwargs,
     ):
         """
         Checks if optimised path for transfer between File location to database exists
-        and if it does, it transfers it and returns true else false.
+        and if it does, it transfers it and returns true else false
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        :param native_support_kwargs: kwargs to be used by native loading command
         """
         raise NotImplementedError
+
+    def create_empty_table(
```

Comment:
Since we already have `create_table` - which creates an empty table using `table.columns`, perhaps we should name this `create_table_with_schema_autodetection`, WDYT?